### PR TITLE
[unit-tests-only] Add unit tests for List component

### PR DESCRIPTION
**Pull request by saw-jan**
## Description
Adds unit  tests for `List` component

## Related Issue
- Part of #5234

## Motivation and Context
- increase unit tests coverage

## How Has This Been Tested?
- test environment: local & CI

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests

## Checklist:
- [ ] Code changes
- [x] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: packages/web-app-files/tests/unit/components/Search/List.spec.js [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+      const filesTable = wrapper.find(selectors.filesTable)
+
+      expect(filesTable.exists()).toBeFalsy()
+      expect(noContentMessage.exists()).toBeTruthy()
+      expect(noContentMessage.text()).toEqual('No resource found')
+    })
+    it('should not show files table', () => {
+      const filesTable = wrapper.find(selectors.filesTable)
+
+      expect(filesTable.exists()).toBeFalsy()
```
Comment (kiranparajuli589) [inline packages/web-app-files/tests/unit/components/Search/List.spec.js:84]: already tested in the scenario just before this
maybe just putting it here will be nice

File: packages/web-app-files/tests/unit/components/Search/List.spec.js [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+    })
+    it('should show files table', () => {
+      const filesTable = wrapper.find(selectors.filesTable)
+      const pagination = wrapper.find(selectors.pagination)
+
+      expect(filesTable.exists()).toBeTruthy()
+      expect(filesTable.props().resources).toMatchObject(files)
+      expect(pagination.exists()).toBeTruthy()
+    })
+    it('should send correct props on list-info component', () => {
```
Comment (kiranparajuli589) [inline packages/web-app-files/tests/unit/components/Search/List.spec.js:112]:
```suggestion
    it('should set correct props on list-info component', () => {
```

File: packages/web-app-files/tests/unit/components/Search/List.spec.js [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+      const filesTable = wrapper.find(selectors.filesTable)
+      await filesTable.vm.$emit('fileClick')
+
+      expect(spyTriggerDefaultAction).toHaveBeenCalledTimes(1)
+    })
+    it('should call "rowMounted" function on "rowMounted" event', async () => {
+      const filesTable = wrapper.find(selectors.filesTable)
+      await filesTable.vm.$emit('rowMounted')
+
+      expect(spyRowMounted).toHaveBeenCalled()
```
Comment (kiranparajuli589) [inline packages/web-app-files/tests/unit/components/Search/List.spec.js:130]:
```suggestion
      expect(spyRowMounted).toHaveBeenCalledTimes(1)
```
will be more informative

File: packages/web-app-files/tests/unit/components/Search/List.spec.js [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+      const filesTable = wrapper.find(selectors.filesTable)
+
+      expect(filesTable.exists()).toBeFalsy()
+      expect(noContentMessage.exists()).toBeTruthy()
+      expect(noContentMessage.text()).toEqual('No search term entered')
+    })
+    it('should not show files table', () => {
+      const filesTable = wrapper.find(selectors.filesTable)
+
+      expect(filesTable.exists()).toBeFalsy()
```
Comment (kiranparajuli589) [inline packages/web-app-files/tests/unit/components/Search/List.spec.js:63]: already tested in the above scenario.

File: packages/web-app-files/tests/unit/components/Search/List.spec.js [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+    beforeEach(() => {
+      wrapper = getWrapper('lorem', files)
+    })
+
+    it('should not show no-content-message component', () => {
+      const noContentMessage = wrapper.find(selectors.noContentMessage)
+
+      expect(noContentMessage.exists()).toBeFalsy()
+    })
+    it('should show files table', () => {
```
Comment (kiranparajuli589) [inline packages/web-app-files/tests/unit/components/Search/List.spec.js:104]:
```suggestion
    it('should show files table with pagination', () => {
```

**Review by dpakach — COMMENTED** [review]
LGTM

File: packages/web-app-files/tests/unit/components/Search/List.spec.js [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+      expect(listInfo.exists()).toBeTruthy()
+      expect(listInfo.props().files).toEqual(files.length)
+      expect(listInfo.props().folders).toEqual(0)
+      expect(listInfo.props().size).toEqual(getTotalSize(files))
+    })
+    it('should call "$_fileActions_triggerDefaultAction" method on "fileClick" event', async () => {
+      const filesTable = wrapper.find(selectors.filesTable)
+      await filesTable.vm.$emit('fileClick')
+
+      expect(spyTriggerDefaultAction).toHaveBeenCalledTimes(1)
```
Comment (kiranparajuli589) [inline packages/web-app-files/tests/unit/components/Search/List.spec.js:107]: also, check if `spy` is called before emitting the event

File: packages/web-app-files/tests/unit/components/Search/List.spec.js [hunk 7]
```diff
@@ -0,0 +1,171 @@
...
+    jest.clearAllMocks()
+  })
+
+  describe.each(['No search term entered', 'No resource found'])('when %s', message => {
+    let wrapper
+    beforeEach(() => {
+      if (message === 'No search term entered') {
+        wrapper = getWrapper()
+      } else {
+        wrapper = getWrapper('epsum')
```
Comment (pascalwengerter) [inline packages/web-app-files/tests/unit/components/Search/List.spec.js:52]: What is `'epsum'`?

**Comment by saw-jan** [comment]
currently CI is failing due to an issue that will be fixed by PR https://github.com/owncloud/web/pull/5833

**Pull request merged**